modular_skyrat/modules/mold/code/mold_controller.dm

```diff
@@ -40,67 +92,57 @@
 	var/our_turf = get_turf(our_core)
 	turfs[our_turf] = TRUE
 	for(var/i in 1 to spread_radius)
-		for(var/tr in turfs)
-			var/turf/T = tr
-			for(var/tr2 in T.atmos_adjacent_turfs)
-				turfs[tr2] = TRUE
-	for(var/tr in turfs)
-		var/turf/T = tr
-		SpawnResin(T)
-		if(T == our_turf)
+		for(var/turf/iterated_turf as anything in turfs)
+			for(var/turf/adjacent_turf as anything in iterated_turf.atmos_adjacent_turfs)
+				turfs[adjacent_turf] = TRUE
+	for(var/turf/iterated_turf as anything in turfs)
+		spawn_resin(iterated_turf)
+		if(iterated_turf == our_turf)
 			continue
 		if(hatcheries_to_spawn && prob(40))
 			hatcheries_to_spawn--
-			SpawnStructureLoc(2, T)
+			spawn_structure_loc(2, iterated_turf)
 		else if(bulbs_to_spawn && prob(40))
 			bulbs_to_spawn--
-			SpawnStructureLoc(1, T)
+			spawn_structure_loc(1, iterated_turf)
 		else if(conditioners_to_spawn && prob(40))
 			conditioners_to_spawn--
-			SpawnStructureLoc(3, T)
+			spawn_structure_loc(3, iterated_turf)
 
-/datum/biohazard_blob_controller/proc/SpawnStructureLoc(index, location)
+/datum/mold_controller/proc/spawn_structure_loc(index, location)
 	var/spawn_type
 	switch(index)
 		if(1)
-			spawn_type = /obj/structure/biohazard_blob/structure/bulb
+			spawn_type = /obj/structure/mold/structure/bulb
 		if(2)
-			spawn_type = /obj/structure/biohazard_blob/structure/spawner
+			spawn_type = /obj/structure/mold/structure/spawner
 		if(3)
-			spawn_type = /obj/structure/biohazard_blob/structure/conditioner
+			spawn_type = /obj/structure/mold/structure/conditioner
 
-	var/struct = new spawn_type(location, blob_type)
+	var/struct = new spawn_type(location, mold_type)
 	other_structures[struct] = TRUE
 	our_core.max_integrity += 10
 	our_core.repair_damage(10)
 	return struct
 
-/datum/biohazard_blob_controller/Destroy()
-	STOP_PROCESSING(SSobj, src)
-	all_resin = null
-	active_resin = null
-	our_core = null
-	other_structures = null
-	return ..()
-
-/datum/biohazard_blob_controller/proc/CoreRetaliated()
+/datum/mold_controller/proc/core_retaliated()
 	structure_progression += PROGRESSION_RETALIATED
 	active_resin.Cut()
-	ActivateAdjacentResinRecursive(get_turf(our_core), 4)
+	activate_adjacent_resin_recursive(get_turf(our_core), 4)
 
-/datum/biohazard_blob_controller/proc/TrySpreadResin(obj/structure/biohazard_blob/resin/spreaded_resin)
+/datum/mold_controller/proc/try_spread_resin(obj/structure/mold/resin/spreaded_resin)
 	. = RESIN_CANT_SPREAD
 	var/turf/ownturf = get_turf(spreaded_resin)
 	if(structure_progression > PROGRESSION_FOR_STRUCTURE)
 		var/forbidden = FALSE
 		for(var/obj/O in ownturf)
-			if(istype(O, /obj/structure/biohazard_blob/structure))
+			if(istype(O, /obj/structure/mold/structure))
 				forbidden = TRUE
 				break
 		if(!forbidden)
 			structure_progression -= PROGRESSION_FOR_STRUCTURE
 			var/random = rand(1,3)
-			SpawnStructureLoc(random, ownturf)
+			spawn_structure_loc(random, ownturf)
```

Comment:
Random what? Could have those two numbers being defines as well, I know it wasn't the goal of this thing but this just makes me sad to look at.